.github/workflows/master_release.yml

```diff
@@ -0,0 +1,52 @@
+name: Master/Release branch workflow
```

Comment:
I'd prefer to merge this file back into `ci.yml` as the sheer amount of repetition that is required creates more stuff to maintain than what we save from modularizing the CI/CD workflows like this.